tests/conftest.py

```diff
@@ -1,79 +1,66 @@
 import os
-import urllib
-from collections.abc import Generator
+from collections.abc import Generator, Iterator
 from multiprocessing import Process
-from time import sleep
 from typing import Any
-from urllib.error import URLError
 
+import httpx
 import pytest
-import uvicorn
 from _pytest.fixtures import SubRequest
 from fastapi.testclient import TestClient
 from playwright.sync_api import Page, Playwright, sync_playwright
-from sqlmodel import Session
-from tad.core.config import settings
+from sqlmodel import Session, SQLModel
+from tad.core.config import Settings, get_settings
 from tad.core.db import get_engine
 from tad.main import app
+from uvicorn.main import run as uvicorn_run
 
 from tests.database_test_utils import DatabaseTestUtils
 
 
-class TestSettings:
-    HTTP_SERVER_SCHEME: str = "http://"
-    HTTP_SERVER_HOST: str = "127.0.0.1"
-    HTTP_SERVER_PORT: int = 8000
-    RETRY: int = 10
-
-
-def run_server() -> None:
-    uvicorn.run(app, host=TestSettings.HTTP_SERVER_HOST, port=TestSettings.HTTP_SERVER_PORT)
-
-
-def wait_for_server_ready(server: Generator[Any, Any, Any]) -> None:
-    for _ in range(TestSettings.RETRY):
-        try:
-            # we use urllib instead of playwright, because we only want a simple request
-            #  not a full page with all assets
-            assert urllib.request.urlopen(server).getcode() == 200  # type: ignore # noqa
-            break
-        # todo (robbert) find out what exception to catch
-        except URLError:  # server was not ready
-            sleep(1)
+@pytest.fixture(scope="module")
+def run_server(request: pytest.FixtureRequest) -> Generator[Any, None, None]:
+    def run_uvicorn(uvicorn: Any) -> None:
```

Comment:
Defining this function within run_server throws a 'can not unpickle object' error in my Python 3.12. It works if you keep the run_uvicorn as separate (root level) method.

Reply:
thanks. i will have a look at python 3.12